flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/table.scala

```diff
@@ -472,61 +339,39 @@ class Table(
     tableEnv.emitToSink(this, configuredSink)
   }
 
-  private def createRenamingProject(exprs: Seq[RexNode]): LogicalProject = {
-
-    val names = exprs.map{ e =>
-      e.getKind match {
-        case SqlKind.AS =>
-          e.asInstanceOf[RexCall].getOperands.get(1)
-            .asInstanceOf[RexLiteral].getValue
-            .asInstanceOf[NlsString].getValue
-        case SqlKind.INPUT_REF =>
-          relNode.getRowType.getFieldNames.get(e.asInstanceOf[RexInputRef].getIndex)
-        case _ =>
-          throw new PlanGenException("Unexpected expression type encountered.")
-      }
-
-    }
-    LogicalProject.create(relNode, exprs.toList.asJava, names.toList.asJava)
-  }
-
   private def checkUniqueNames(exprs: Seq[Expression]): Unit = {
     val names: mutable.Set[String] = mutable.Set()
 
     exprs.foreach {
-      case n: Naming =>
+      case n: Alias =>
         // explicit name
         if (names.contains(n.name)) {
-          throw new IllegalArgumentException(s"Duplicate field name $n.name.")
+          throw new ValidationException(s"Duplicate field name $n.name.")
         } else {
           names.add(n.name)
         }
       case u: UnresolvedFieldReference =>
         // simple field forwarding
         if (names.contains(u.name)) {
-          throw new IllegalArgumentException(s"Duplicate field name $u.name.")
+          throw new ValidationException(s"Duplicate field name $u.name.")
         } else {
           names.add(u.name)
         }
       case _ => // Do nothing
     }
   }
 
+  @inline protected def validate(logicalNode: => LogicalNode): Table = {
+    new Table(tableEnv, logicalNode.validate(tableEnv))
```

Comment:
Would it be OK to remove this method and move the `new Table` call to each API method? 
I think this would make the code easier to read.